sssd_test_framework/hosts/ipa.py

```diff
@@ -45,12 +47,46 @@ def __init__(self, *args, **kwargs) -> None:
         self.adminpw: str = self.config.get("adminpw", "Secret123")
         """Password of the admin user, defaults to ``Secret123``."""
 
+        self._features: dict[str, bool] | None = None
+
         # Additional client configuration
         self.client.setdefault("id_provider", "ipa")
         self.client.setdefault("access_provider", "ipa")
         self.client.setdefault("ipa_server", self.hostname)
         self.client.setdefault("dyndns_update", False)
 
+    @property
+    def features(self) -> dict[str, bool]:
+        """
+        Features supported by the host.
+        """
+        self.kinit()
+
+        if self._features is not None:
+            return self._features
+
+        self.logger.info(f"Detecting features on {self.hostname}")
+
+        result = self.ssh.run(
+            """
+            set -ex
+
+            [ -f "/usr/libexec/sssd/passkey_child" ] && ipa help user | \
+                grep user-add-passkey 1> /dev/null && echo "passkey" || :
```

Comment:
nitpick but this would help readability

```
"""
set -ex

[ -f "/usr/libexec/sssd/passkey_child" ] && \
ipa help user | grep user-add-passkey 1> /dev/null && \
echo "passkey" || :
""",
```

Reply:
Done